library-best-practices/JLBP-9.md

```diff
@@ -0,0 +1,10 @@
+[JLBP-9] Ensure consumers always use a higher or equal minimum Java version
+---------------------------------------------------------------------------
+
+If a library B requiring Java 1.7 depends on a library A requiring Java 1.7, and
+library A bumps its minimum version to Java 1.8, then library B cannot pick up
+the new version of library A until it also bumps its own minimum Java
+version. This can result in a situation where the ecosystem is stuck on older
+library versions, preventing the pickup of important things like security
```

Comment:
Worth calling out that it also leads to diamond-dependency problems -- ones where there's _no_ version that works for all consumers?

Reply:
Good point, I'll add that in.